tests/measurements/test_sample.py

```diff
@@ -500,6 +499,38 @@ def circuit(x):
     )
 
 
+@pytest.mark.jax
+@pytest.mark.parametrize(
+    "obs",
+    [
+        # Single observables
+        (qml.PauliX(0)),
+        (qml.PauliY(0)),
+        (qml.PauliZ(0)),
+        (qml.Hadamard(0)),
+        (qml.Identity(0)),
+    ],
+)
+def test_jitting_with_sampling_on_different_observables(obs):
+    """Test case covering bug in Issue #5369. Sampling should be jit-able
+    when sampling occurs and obs is not None. The bug was occurring due mismatched
+    types on qml.sample(obs).numeric_type and obs.eigvals."""
```

Comment:
Nice to have this context. Is this something we do? Retroactively explain why code was failing that is no longer there.

Reply:
Honestly no, not really, maybe I should shorten it for consistency

Reply:
I think that would be good. We should make a concious decision if we wanna start doing this :)